### PR TITLE
Setup infrastructure

### PR DESCRIPTION
**Pull request by ghost**
* Add infrastructure overview schema
* Add backend swagger
* Configure Caddy gateway
* Configure Docker composer
* Add CD scripts

### CODE REVIEW

**Review by AHGIJMKLKKZNPJKQR — APPROVED** [review]
Looks clean, might want to add the constants from `agartex-*/src/constants.rs` to `agartex-*.env`, but it's not necessary.

**Review by katzper-michno — CHANGES_REQUESTED** [review]
Can we allow liberal CORS for now? To allow everything

**Comment by ghost** [comment]
CORS is disabled

**Comment by katzper-michno** [comment]
Exactly, and my question is whether we can have it >>>>>>>>ENABLED FOR EVERY POSSIBLE ORIGIN<<<<<<<<<<!!!!!!!!!!

**Comment by ghost** [comment]
Sorry, misunderstood, I thought it supposed to be as you desire but it is not. Working on it

**Comment by ghost** [comment]
Roughly tested, liberal CORS is temporarily enabled (allow all origins), blocked by https://github.com/Agar-OSS/agartex-authentication/pull/3

File: gateway/Caddyfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+	handle * {
+		forward_auth http://agartex-authentication:3100 {
+			uri /users
+			copy_headers X-User-Id
+		}
+
+		handle /compile {
+			reverse_proxy http://agartex-compilation:3200
+		}
```
Comment (AHGIJMKLKKZNPJKQR) [inline gateway/Caddyfile]: Shouldn't be protected

Comment (AHGIJMKLKKZNPJKQR) [reply gateway/Caddyfile]: Also port should be 3300

File: gateway/Caddyfile [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+		handle /collaboration {
+			rewrite * /
+			reverse_proxy http://agartex-colaboration:3300
```
Comment (AHGIJMKLKKZNPJKQR) [inline gateway/Caddyfile]: Port should be 3400

**Review by AHGIJMKLKKZNPJKQR — CHANGES_REQUESTED** [review]

File: gateway/Caddyfile (hunk 1 above)
Comment (ghost) [reply gateway/Caddyfile]: It's for demo purpose, I keep it protected

**Review by AHGIJMKLKKZNPJKQR — APPROVED** [review]

**Comment by ghost** [comment]
More restrictive CORS is now enabled to allow requests with cookies

File: gateway/Caddyfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+	handle /sessions {
+		reverse_proxy http://agartex-authentication:3100
+	}
+
+	# Protected routes
+
+	handle * {
+		forward_auth http://agartex-authentication:3100 {
+			uri /users
```
Comment (AHGIJMKLKKZNPJKQR) [inline gateway/Caddyfile]: https://github.com/Agar-OSS/agartex-authentication/pull/8
This needs to now be `/sessions`

**Review by AHGIJMKLKKZNPJKQR — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Issue link: https://github.com/Agar-OSS/agartex-service-core/issues/2

**Review by katzper-michno — APPROVED** [review]

**Review by AHGIJMKLKKZNPJKQR — APPROVED** [review]

**Pull request merged**